Index: installer/chromeos_legacy.cc
===================================================================
--- installer.orig/chromeos_legacy.cc
+++ installer/chromeos_legacy.cc
@@ -252,12 +252,13 @@ bool RunLegacyUBootPostInstall(const Ins
 
   result &= CopyBootFile(install_config,
                          "boot/boot-" + install_config.slot + ".scr.uimg",
-                         "u-boot/boot.scr.uimg");
+                         "boot/boot.scr.uimg");
+#if 0
   result &= CopyBootFile(
       install_config, "boot/uEnv." + install_config.slot + ".txt", "uEnv.txt");
   result &= CopyBootFile(install_config, "boot/MLO", "MLO");
   result &= CopyBootFile(install_config, "boot/u-boot.img", "u-boot.img");
-
+#endif
   return result;
 }
 
Index: installer/chromeos_postinst.cc
===================================================================
--- installer.orig/chromeos_postinst.cc
+++ installer/chromeos_postinst.cc
@@ -116,12 +116,12 @@ bool DetectBiosType(BiosType* bios_type)
 
 bool KernelConfigToBiosType(const string& kernel_config, BiosType* type) {
   if (kernel_config.find("cros_secure") != string::npos) {
-    *type = BiosType::kSecure;
+    *type = BiosType::kUBoot;
     return true;
   }
 
   if (kernel_config.find("cros_legacy") != string::npos) {
-#ifdef __arm__
+#if defined(__arm__) || defined(__aarch64__)
     // The Arm platform only uses U-Boot, but may set cros_legacy to mean
     // U-Boot without our secure boot modifications.
     *type = BiosType::kUBoot;
