Index: installer/chromeos-postinst
===================================================================
--- installer.orig/chromeos-postinst
+++ installer/chromeos-postinst
@@ -37,6 +37,12 @@ cleanup() {
   done
 }
 
+get_android_version() {
+  local lsbfile=$1
+  local version=$(cat $lsbfile | grep CHROMEOS_ARC_VERSION)
+  echo ${version#*=}
+}
+
 main() {
   cd "${INSTALL_ROOT}" || exit 1
 
@@ -50,7 +56,9 @@ main() {
       mount --make-slave "./${d}"
     done
   fi
-
+  if [ "$(get_android_version /etc/lsb-release)" -ne "$(get_android_version etc/lsb-release)" ]; then
+    chroot . /usr/sbin/mark_clean_overlay.sh $target_part_num $root_dev
+  fi
   local install_dev="$1"
   shift
   chroot . /usr/bin/cros_installer --type="postinst" \
Index: installer/chromeos-install.sh
===================================================================
--- installer.orig/chromeos-install.sh
+++ installer/chromeos-install.sh
@@ -558,6 +558,8 @@ install_stateful() {
     #
     touch "${TMPMNT}"/.developer_mode
   fi
+  # avoid modify the gpt table
+  touch ${TMPMNT}/.autoexpanded
 
   if [ -n "${IS_RECOVERY_INSTALL-}" ] ; then
     # This is a recovery install; write some recovery metrics to the stateful
