diff --git a/installer/chromeos_legacy.cc b/installer/chromeos_legacy.cc
index 2ebc7e9d51ed..d3b80a271d07 100644
--- a/installer/chromeos_legacy.cc
+++ b/installer/chromeos_legacy.cc
@@ -250,6 +250,8 @@ bool RunLegacyUBootPostInstall(const InstallConfig& install_config) {
   bool result = true;
   LOG(INFO) << "Running LegacyUBootPostInstall.";
 
+  result &= CopyBootFile(install_config,
+                         "boot/boot.scr.uimg", "boot/boot.scr.uimg");
   result &= CopyBootFile(install_config,
                          "boot/boot-" + install_config.slot + ".scr.uimg",
                          "u-boot/boot.scr.uimg");
diff --git a/installer/chromeos_postinst.cc b/installer/chromeos_postinst.cc
index 007403fc76f3..de823b5fca75 100644
--- a/installer/chromeos_postinst.cc
+++ b/installer/chromeos_postinst.cc
@@ -135,8 +135,9 @@ bool KernelConfigToBiosType(const string& kernel_config, BiosType* type) {
   }
 
   // No recognized bios type was found
-  LOG(ERROR) << "No recognized cros_XXX bios option on kernel command line.";
-  return false;
+  //LOG(ERROR) << "No recognized cros_XXX bios option on kernel command line.";
+  *type = BiosType::kUBoot;
+  return true;
 }
 
 namespace {
