diff --git a/installer/chromeos-postinst b/installer/chromeos-postinst
index 9fbae9c195..34dd3e1c82 100755
--- a/installer/chromeos-postinst
+++ b/installer/chromeos-postinst
@@ -29,7 +29,34 @@ cleanup() {
   done
 }
 
+get_cur_dev() {
+  mount |grep ${INSTALL_ROOT} |awk '{print $1}'
+}
+
+get_disk_dev() {
+  local disk_dev=$(get_cur_dev)
+  disk_dev=${disk_dev%[0-9]}
+  disk_dev=${disk_dev%[0-9]}
+  echo ${disk_dev%p}
+}
+
+get_part_num() {
+  local disk_dev=$(get_cur_dev)
+  echo ${disk_dev##*[a-z]}
+}
+
+get_android_version() {
+  local lsbfile=$1
+  local version=$(cat $lsbfile | grep CHROMEOS_ARC_VERSION)
+  echo ${version#*=}
+}
+
 main() {
+  local target_disk=$(get_disk_dev)
+  local target_part_num=$(get_part_num)
+  # mount efi of switch may fail on low speed storage, call sync first
+  sync
+  ${INSTALL_ROOT}/usr/sbin/switch_root.sh $target_part_num $target_disk
   cd "${INSTALL_ROOT}" || exit 1
 
   trap cleanup EXIT
@@ -40,6 +65,10 @@ main() {
     mount --make-slave "./${d}"
   done
 
+  if [ "$(get_android_version /etc/lsb-release)" -ne "$(get_android_version etc/lsb-release)" ]; then
+    chroot . /usr/sbin/mark_clean_overlay.sh $target_part_num $root_dev
+  fi
+
   local install_dev="$1"
   shift
   chroot . /usr/bin/cros_installer --type="postinst" \
